power-control-x86/src/power_control.cpp

```diff
@@ -144,18 +163,57 @@ static int getGPIOValue(const std::string& name)
     return value;
 }
 
+static void setFrontpanelPowerLed(bool led)
+{
+    int fd = -1;
+    char i2c_devname[FILEPATHSIZE];
+    int portRead, portWrite, dataRead, dataWrite;
+
+    // check volcano FP
```

Comment:
This logic needs to be broken down.

//create a new function i2ccommandLed(char *i2c_devname) or similar

setFrontpanelPowerLed()
{
   // Attempt bus 1
i2ccommandLed(bus1);
   // Attempt bus 2
i2ccommandLed(bus2);
}

Ideally, we would implement as virtual void setFrontpanelPowerLed();
and override setFrontpanelPowerLed in individual platform files, but this cannot be done quick. Hence the above approach.